### ISSUE TITLE
Tests failed with System.AccessViolationException: Attempted to read or write protected memory. This is often an indication that other memory is corrupt

### ISSUE THREAD

**Issue by VincentBu** [dotnet/runtime#52494]
Run: [runtime-coreclr crossgen2 20210506.1](https://dev.azure.com/dnceng/public/_build/results?buildId=1127214&view=ms.vss-test-web.build-test-results-tab&runId=34255810&resultId=100442&paneView=debug)

Failed tests on windows:
```
- Loader\\classloader\\generics\\Instantiation\\Positive\\NestedInterface07\\NestedInterface07.cmd
- JIT\\Generics\\Instantiation\\Interfaces\\struct03\\struct03.cmd
- Loader\\classloader\\generics\\Instantiation\\Positive\\NestedInterface08\\NestedInterface08.cmd
- JIT\\Generics\\Instantiation\\Interfaces\\struct02\\struct02.cmd
```

**One of Error messages:**
```
Could Not Find C:\h\w\AC290951\w\B69709B8\e\Loader\classloader\generics\Instantiation\Positive
estedInterface07
estedInterface07.dll.rsp Fatal error. System.AccessViolationException: Attempted to read or write protected memory. This is often an indication that other memory is corrupt.
Return code: 1 Raw output file: C:\h\w\AC290951\w\B69709B8\e\Loader\classloader\Reports\Loader.classloader\generics\Instantiation\Positive
estedInterface07
estedInterface07.output.txt Raw output: BEGIN EXECUTION NestedInterface07.dll 1 file(s) copied. Response file: C:\h\w\AC290951\w\B69709B8\e\Loader\classloader\generics\Instantiation\Positive
estedInterface07
estedInterface07.dll.rsp C:\h\w\AC290951\w\B69709B8\e\Loader\classloader\generics\Instantiation\Positive
estedInterface07\IL-CG2
estedInterface07.dll -o:C:\h\w\AC290951\w\B69709B8\e\Loader\classloader\generics\Instantiation\Positive
estedInterface07
estedInterface07.dll --targetarch:x64 --verify-type-and-field-layout -r:C:\h\w\AC290951\p\System..dll -r:C:\h\w\AC290951\p\Microsoft..dll -r:C:\h\w\AC290951\p\mscorlib.dll -r:C:\h\w\AC290951\p
etstandard.dll -O " "dotnet" "C:\h\w\AC290951\p\crossgen2\crossgen2.dll" @"C:\h\w\AC290951\w\B69709B8\e\Loader\classloader\generics\Instantiation\Positive
estedInterface07
estedInterface07.dll.rsp" -r:C:\h\w\AC290951\w\B69709B8\e\Loader\classloader\generics\Instantiation\Positive
estedInterface07\IL-CG2*.dll" Emitting R2R PE file: C:\h\w\AC290951\w\B69709B8\e\Loader\classloader\generics\Instantiation\Positive
estedInterface07
estedInterface07.dll "C:\h\w\AC290951\p\corerun.exe" NestedInterface07.dll Expected: 100 Actual: -1073741819 END EXECUTION - FAILED FAILED Test Harness Exitcode is : 1 To run the test:
set CORE_ROOT=C:\h\w\AC290951\p C:\h\w\AC290951\w\B69709B8\e\Loader\classloader\generics\Instantiation\Positive
estedInterface07
estedInterface07.cmd Expected: True Actual: False

Stack trace
   at Loader_classloader._generics_Instantiation_Positive_NestedInterface07_NestedInterface07_._generics_Instantiation_Positive_NestedInterface07_NestedInterface07_cmd()

```

Failed tests on Linux and OSX:
```

- Loader/classloader/generics/Instantiation/Positive/NestedInterface07/NestedInterface07.sh
- JIT/Generics/Instantiation/Interfaces/struct03/struct03.sh
- Loader/classloader/generics/Instantiation/Positive/NestedInterface08/NestedInterface08.sh
- JIT/Generics/Instantiation/Interfaces/struct02/struct02.sh

```

**One of Error messages:**
```
/datadisks/disk1/work/ADB30977/w/9F5C08F9/e/Loader/classloader/generics/Instantiation/Positive/NestedInterface07/NestedInterface07.sh: line 254: -r:/datadisks/disk1/work/ADB30977/w/9F5C08F9/e/Loader/classloader/generics/Instantiation/Positive/NestedInterface07/IL-CG2/*.dll: No such file or directory
rm: cannot remove '/datadisks/disk1/work/ADB30977/w/9F5C08F9/e/Loader/classloader/generics/Instantiation/Positive/NestedInterface07/NestedInterface07.dll.rsp': No such file or directory
/datadisks/disk1/work/ADB30977/w/9F5C08F9/e/Loader/classloader/generics/Instantiation/Positive/NestedInterface07/NestedInterface07.sh: line 375: 12065 Segmentation fault      (core dumped) $LAUNCHER $ExePath "${CLRTestExecutionArguments[@]}"

Return code:      1
Raw output file:      /datadisks/disk1/work/ADB30977/w/9F5C08F9/e/Loader/classloader/Reports/Loader.classloader/generics/Instantiation/Positive/NestedInterface07/NestedInterface07.output.txt
Raw output:
BEGIN EXECUTION
in takeLock
/datadisks/disk1/work/ADB30977/w/9F5C08F9/e/Loader/classloader/generics/Instantiation/Positive/NestedInterface07/IL-CG2/NestedInterface07.dll
Response file: /datadisks/disk1/work/ADB30977/w/9F5C08F9/e/Loader/classloader/generics/Instantiation/Positive/NestedInterface07/NestedInterface07.dll.rsp
/datadisks/disk1/work/ADB30977/w/9F5C08F9/e/Loader/classloader/generics/Instantiation/Positive/NestedInterface07/IL-CG2/NestedInterface07.dll
-o:/datadisks/disk1/work/ADB30977/w/9F5C08F9/e/Loader/classloader/generics/Instantiation/Positive/NestedInterface07/NestedInterface07.dll
-r:/datadisks/disk1/work/ADB30977/p/System..dll
-r:/datadisks/disk1/work/ADB30977/p/Microsoft..dll
-r:/datadisks/disk1/work/ADB30977/p/mscorlib.dll
--verify-type-and-field-layout
--targetarch:x64
-O
Running CrossGen2:  dotnet /datadisks/disk1/work/ADB30977/p/crossgen2/crossgen2.dll @/datadisks/disk1/work/ADB30977/w/9F5C08F9/e/Loader/classloader/generics/Instantiation/Positive/NestedInterface07/NestedInterface07.dll.rsp

Emitting R2R PE file: /datadisks/disk1/work/ADB30977/w/9F5C08F9/e/Loader/classloader/generics/Instantiation/Positive/NestedInterface07/NestedInterface07.dll
in ReleaseLock
/datadisks/disk1/work/ADB30977/p/corerun NestedInterface07.dll ''
Gathering state for process 12065 corerun
Writing minidump with heap to file /home/helixbot/dotnetbuild/dumps/coredump.12065.dmp
Written 52867072 bytes (12907 pages) to core file
Dump successfully written
Expected: 100
Actual: 139
END EXECUTION - FAILED
Test Harness Exitcode is : 1
To run the test:

set CORE_ROOT=/datadisks/disk1/work/ADB30977/p
/datadisks/disk1/work/ADB30977/w/9F5C08F9/e/Loader/classloader/generics/Instantiation/Positive/NestedInterface07/NestedInterface07.sh
Expected: True
Actual:   False


Stack trace
   at Loader_classloader._generics_Instantiation_Positive_NestedInterface07_NestedInterface07_._generics_Instantiation_Positive_NestedInterface07_NestedInterface07_sh()
```

**Comment by BruceForstall** [comment]
@dotnet/crossgen-contrib

**Comment by trylek** [comment]
Likely duplicate of: https://github.com/dotnet/runtime/issues/52483

**Comment by AndyAyersMS** [comment]
Agree this was a dup, addressed by #52605. These tests are passing in 20210512.1.
